src/directory/footer.rs

```diff
@@ -120,7 +148,7 @@ impl BinarySerializable for VersionedFooter {
             VersionedFooter::V1 { crc32, compression } => {
                 // Serializes a valid `VersionedFooter` or panics if the version is unknown
                 // [   version    |   crc_hash  | compression_mode ]
-                // [    0..4      |     4..8    |  variable ]
```

Comment:
What was wrong with the variable compression mode?

Reply:
I thought it would be simpler, if we fix the compression_mode as a 4-char string. 

In the future, when compile-time enums are supported (and if we add more compression modes), we can replace the compression string with an enum, which will (de)serialise to a u32 value.

Reply:
I don't think it is simpler.